activerecord/test/cases/associations/eager_test.rb

```diff
@@ -968,8 +970,38 @@ def test_preconfigured_includes_with_has_one
   end
 
   def test_eager_association_with_scope_with_joins
-    assert_nothing_raised do
```

Comment:
I think that this is a weaker test than the one introduced. By putting an outcome that can only happen from a joins (duplicates), it shows that the joins work.